core/src/main/scala/dagr/core/cmdline/parsing/DagrCommandLineParser.scala

```diff
@@ -79,117 +47,25 @@ private[cmdline] class DagrCommandLineParser(val commandLineName: String, val in
     * and then parses all of the arguments and constructs the necessary objects.
     */
   def parse(args: Array[String], packageList: List[String]): Option[(DagrCoreMain, Pipeline)] = {
-    /* How Usage messages are intended to be printed. The following rules should be applied, in precedence order.
-     1. If dagr arguments are missing, always print out the dagr usage.
-     2. If no task name is given, just print the dagr usage and the list of available WFs.
-     3. If the task is specified and there are missing task arguments, always print out the task usage.
-     4. If "-h" is used in dagr's options, print out just the dagr usage.
-     5. If "-h" is used in the task's options, print out just the task usage.
-     6. If "-H' is used in the task's options, print out both the dagr and task usage.
-     */
 
-    // Parse the args for Dagr
-    val mainClass = classOf[DagrCoreMain]
-    val dagrArgParser = new CommandLineParser(mainClass) {
-      override protected def standardUsagePreamble: String = {
-        s"$KRED$UsagePrefix $KBLDRED$commandLineName$KNRM$KRED [$commandLineName arguments] -- [Task Name] [task arguments]$KNRM\n\n"
-      }
-      override protected def targetName: String = Configuration.commandLineName
+    val parser = new CommandLineParser[Pipeline](commandLineName=commandLineName) {
+      override def genericClpNameOnCommandLine: String = "pipeline"
     }
 
-    val print : (String => Unit) = System.err.println
-    val errorMessageBuilder: mutable.StringBuilder = new StringBuilder()
-    val (dagrArgs, pipelineArgs) = splitArgs(args, packageList, Seq(classOf[DagrCoreMain]), errorMessageBuilder, includeHidden = this.includeHidden)
-
-    /** A brief developer note about the precedence of printing error messages and usages.
-      *
-      * 1. Complain if dagr options are mis-specified.
-      * 2. Print the dagr-only help message if --help is used before a pipeline name.
-      * 3. Print the version if --version is used before a pipeline name.
-      * 4. Complain if (a) no pipeline name is given, or (b) it is not recognized.
-      * 5. Complain if pipeline options are mis-specified.
-      * 6. Print the dagr-and-pipeline help message if --help is used after a pipeline name.
-      * 7. Print the version if --version is used after a pipeline name.
-      * 8. ... execute dagr ...
-      * */
+    // Load any Dagr scripts and add them to the classpath.
+    val mainBlock = (dagr: DagrCoreMain) => { loadScripts(clp=dagr); Unit }
```

Comment:
This caused me to unravel a few things, I don't know why I had `Unit.type` everywhere.